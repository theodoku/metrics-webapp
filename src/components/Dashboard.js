import PropTypes from "prop-types";
import "./Dashboard.css";

function Dashboard(props) {
  const { map, name, population } = props;

  return (
    <section className="Dashboard">
      <img src={map} alt={name} className="dashboard-map" />
      <article className="map-container">
        <h2 className="dash-title">{name}</h2>
        <span className="pop">{population}</span>
      </article>
    </section>
  );
}

Dashboard.propTypes = {
  name: PropTypes.string,
  population: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  map: PropTypes.string,
};

Dashboard.defaultProps = {
  name: "",
  population: "",
  map: "",
};

export default Dashboard;
