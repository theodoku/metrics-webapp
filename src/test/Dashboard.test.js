import React from 'react';
import { render } from '@testing-library/react';
import Dashboard from '../components/Dashboard.js';

describe('Dashboard component', () => {
  it('renders the component with props', () => {
    const props = {
      name: 'Test City',
      population: 100000,
      map: 'https://test.com/map.png',
    };

    const { container } = render(
      <Dashboard
        name={props.name}
        population={props.population}
        map={props.map}
      />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
