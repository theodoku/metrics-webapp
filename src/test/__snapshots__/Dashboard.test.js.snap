// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component renders the component with props 1`] = `
<section
  class="Dashboard"
>
  <img
    alt="Test City"
    class="dashboard-map"
    src="https://test.com/map.png"
  />
  <article
    class="map-container"
  >
    <h2
      class="dash-title"
    >
      Test City
    </h2>
    <span
      class="pop"
    >
      100000
    </span>
  </article>
</section>
`;
